File: frontend-app/profile-page/src/model/BookingResponseDTO.ts
import { LocationResponseDTO } from './LocationResponseDTO';
import { BookingStateDTO } from './BookingStateDTO';

export class BookingResponseDTO {
    id: string;
    userId: string;
    vehicleId: string;
    receiptDate: Date;
    returnDate: Date;
    location: LocationResponseDTO;
    bookingState: BookingStateDTO;
    totalCost: number;

    constructor(
        id: string,
        userId: string,
        vehicleId: string,
        receiptDate: Date,
        returnDate: Date,
        location: LocationResponseDTO,
        bookingState: BookingStateDTO,
        totalCost: number
    ) {
        this.id = id;
        this.userId = userId;
        this.vehicleId = vehicleId;
        this.receiptDate = receiptDate;
        this.returnDate = returnDate;
        this.location = location;
        this.bookingState = bookingState;
        this.totalCost = totalCost;
    }
}

File: api/booking-service/src/main/java/com/carrental/bookingservice/model/dto/BookingResponseDTO.java
package com.carrental.bookingservice.model.dto;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDate;

@Data
public class BookingResponseDTO {

  private Long id;

  private String userId;

  private String vehicleId;

  @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd")
  private LocalDate receiptDate;

  @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd")
  private LocalDate returnDate;

  private LocationResponseDTO location;

  private BookingStateDTO bookingState;

  private BigDecimal totalCost;
}
